pkg/controller/dnsendpoint/dnsendpoint_controller.go

```diff
@@ -162,7 +205,13 @@ func (r *ReconcileDNSEndpoint) Reconcile(request reconcile.Request) (reconcile.R
 	}
 
 	desiredNameServers := sets.NewString(instance.Status.NameServers...)
-	rootDomain, currentNameServers := r.nameServerScraper.GetEndpoint(domain)
+	_, currentNameServers := scraper.GetEndpoint(fullDomain) // remove returning root???
+
+	nsQuery, ok := r.nameServerQueryMap[rootDomain]
+	if !ok {
+		dnsLog.Warn("no name server query client found for domain")
```

Comment:
This should be an error-level log entry. It is a failure of the validation to get a domain that is not covered by one of the managed domains.